logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/JoinConversationViaCodeUseCase.kt

```diff
@@ -0,0 +1,67 @@
+package com.wire.kalium.logic.feature.conversation
+
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationGroupRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.toModel
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.network.api.base.authenticated.conversation.ConversationMemberAddedResponse
+
+/**
+ * Use case for joining a conversation via a code invite code.
+ * the param can be obtained from the deep link
+ * @param code The code of the conversation to join.
+ * @param key The key of the conversation to join.
+ * @param domain optional domain of the conversation to join.
+ */
+class JoinConversationViaCodeUseCase internal constructor(
+    private val conversionsGroupRepository: ConversationGroupRepository,
+    private val selfUserId: UserId
+) {
+    suspend operator fun invoke(code: String, key: String, domain: String?): Result =
+        conversionsGroupRepository.joinViaInviteCode(code, key, null)
+            .fold({ failure ->
+                Result.Failure(failure)
+            }, { response ->
+                when (response) {
+                    is ConversationMemberAddedResponse.Changed -> onConversationChanged(response)
+                    ConversationMemberAddedResponse.Unchanged -> onConversationUnChanged(code, key, domain)
+                }
+            })
+
+    private fun onConversationChanged(response: ConversationMemberAddedResponse.Changed): Result =
+        Result.Success.Changed(response.event.qualifiedConversation.toModel())
+
+    private suspend fun onConversationUnChanged(
+        code: String,
+        key: String,
+        domain: String?
+    ): Result =
+        conversionsGroupRepository.fetchLimitedInfoViaInviteCode(code, key)
+            .fold({
+                Result.Success.Unchanged(null)
```

Comment:
This endpoint has 4 different error responses (400, 403, 404, 409), shouldn't we differentiate them?

Reply:
At this point, it is not relevant, here it is called only to get the ID when the user is already a member, and since the user is already joined (the join is called previously) all the other error cases will not be applicable here.

Reply:
Atm i am working on another use case to check the code (before joining) in that case yes i will need to parse the correct errors